src/crypto/public_key/elliptic/curves.js

```diff
@@ -205,13 +223,20 @@ class Curve {
   }
 }
 
-async function generate(curve) {
+async function generate(curve, plugin) {
   const BigInteger = await util.getBigInteger();
 
   curve = new Curve(curve);
-  const keyPair = await curve.genKeyPair();
-  const Q = new BigInteger(keyPair.publicKey).toUint8Array();
-  const secret = new BigInteger(keyPair.privateKey).toUint8Array('be', curve.payloadSize);
+  const keyPair = await curve.genKeyPair(plugin);
+  let Q;
+  let secret;
+  if (curve.name === 'webcrypt_p256') { //web
+    Q = keyPair.publicKey;
+    secret = keyPair.privateKey;
+  } else {
+    Q = new BigInteger(keyPair.publicKey).toUint8Array();
+    secret = new BigInteger(keyPair.privateKey).toUint8Array('be', curve.payloadSize);
+  }
```

Comment:
Is this change necessary? To be honest, I'm also not sure why the last two lines are needed, they seem like a no-op (unless the public key has leading zeros, or the private key has missing leading zeros). Maybe we can check where it's necessary and do it only there.

Reply:
It is needed for p521 (maybe also some brainpool curve, not sure)

Reply:
It was not working for me before, but now does. I can't pinpoint the cause right now - perhaps the plugin.generate implementation was missing proper data in its result? Anyway, reverted that change.